biome.jsonc

```diff
@@ -194,34 +189,16 @@
             ]
           }
         },
-        "useNumberNamespace": "off",
-        "useSingleVarDeclarator": "off",
-        "useTemplate": "off",
-        "noNamespace": "error",
-        "useAsConstAssertion": "error"
+        "noNamespace": "error"
       },
       "suspicious": {
-        "noAssignInExpressions": "error",
-        "noAsyncPromiseExecutor": "off",
+        // `void` as type is useful in our case when used as generic constraint e.g. K extends number | void
         "noConfusingVoidType": "off",
-        "noConsoleLog": "error",
-        "noDoubleEquals": "off",
-        "noDuplicateTestHooks": "off",
-        "noExplicitAny": "error",
-        "noExportsInTest": "off",
-        "noFallthroughSwitchClause": "off",
-        "noGlobalIsFinite": "off",
-        "noGlobalIsNan": "off",
+        // There are lot of cases in our code that need a separate PR to enable this rule
```

Comment:
worth discussion on this one. Afaik, typescript can infer the type most of the time so its not really necessary.

Reply:
Done